samza-core/src/main/scala/org/apache/samza/storage/TaskStorageBackupManager.scala

```diff
@@ -22,11 +22,31 @@ package org.apache.samza.storage
 import org.apache.samza.checkpoint.CheckpointId
 import org.apache.samza.system.SystemStreamPartition
 
-trait TaskStorageManager {
+trait TaskStorageBackupManager {
```

Comment:
I looked a bit more and feel this interface is currently overloaded with responsibilities. For the most part it does only commit handling for stores related to a task and most of the management moved to `ContainerStorageManager` but exposing the store through `getStore` lingers around. 

> If you can take a crack at removing `getStore` method off this interface that will good. `TaskInstance` can get store handles from CSM instead or just directly gets the list of stores associated with it and eliminates CSM from the picture are few options to consider.

The above is not a blocker but do create a follow up if you aren't planning to do.

Given that context, I'd suggest to rename this interface to `TaskCommitManager` and not roll the backup responsibilities into it rather keep it separate and use composition within the commit manager. By this, you will potentially have flexibility to reuse some of the shared commit logic across the types (kafka, remote store, etc) but still keep the back/upload part isolated to its own implementation without have a single class that manages both these.

Let me know your thoughts?

Reply:
I have removed getStore and added the TaskCommitManager, let me know what you think